R/plot.SMCMallows.R

```diff
@@ -0,0 +1,127 @@
+#' to discard as burn-in. Defaults to \code{model_fit$burnin}, and must be
+#' provided if \code{model_fit$burnin} does not exist. See
+#' [assess_convergence()]
+#' @param parameter Character string defining the parameter to plot. Available
+#' options are \code{"alpha"} and \code{"rho"}.
+#' @param time Integer determining the update slice to plot
+#' @param C Number of cluster
+#' @param colnames A vector of item names. If NULL, generic names are generated
+#' for the items in the ranking.
+#' @param items Either a vector of item names, or a vector of indices. If NULL,
+#' five items are selected randomly.
+#' @param ... Other arguments passed to [base::plot()] (not used).
+#' @return A plot of the posterior distributions
+#' @author Waldir Leoncio
+#' @export
+#' @example /inst/examples/plot.SMCMallows_example.R
+plot.SMCMallows <- function(x, nmc = nrow(x$rho_samples[, 1, ]), burnin = 0,
+  parameter = "alpha", time = ncol(x$rho_samples[, 1, ]), C = 1,
+  colnames = NULL, items = NULL, ...) {
+
+  if (parameter == "alpha") {
+    output <- x$alpha_samples[, time]
+    plot_alpha_posterior(output, nmc, burnin)
+  } else if (parameter == "rho") {
+    output <- x$rho_samples[, , time]
+    plot_rho_posterior(output, nmc, burnin, C, colnames, items)
+  } else {
+    stop("parameter must be either 'alpha' or 'rho'.")
+  }
+}
+
+#' @title Plot Alpha Posterior
+#' @description posterior for alpha
+#' @inheritParams compute_posterior_intervals_rho
+#' @export
+#' @author Anja Stein
+#'
+# AS: if you remove the verbose input variable, then the function will be consistent
+# with the other plot functions(they all print when verbose=FALSE, but this function doesn't.)
+# `plot_rho_heatplot` doesn't require the variable `verbose`,
+# so I'm not sure if this function does to plot the density of alpha
+plot_alpha_posterior <- function(output, nmc, burnin) {
+  alpha_samples_table <- data.frame(iteration = 1:nmc, value = output)
+
+  plot_posterior_alpha <- ggplot2::ggplot(alpha_samples_table, ggplot2::aes_(x = ~value)) +
+    ggplot2::geom_density() +
+    ggplot2::xlab(expression(alpha)) +
+    ggplot2::ylab("Posterior density") +
+    ggplot2::ggtitle(label = "Implemented SMC scheme") +
+    ggplot2::theme(plot.title = ggplot2::element_text(hjust = 0.5))
+
+  print(plot_posterior_alpha)
+}
+
+#' @title Plot the posterior for rho for each item
+#' @param output input
+#' @param nmc Number of Monte Carlo samples
+#' @param burnin A numeric value specifying the number of iterations
+#' to discard as burn-in. Defaults to \code{model_fit$burnin}, and must be
+#' provided if \code{model_fit$burnin} does not exist. See \code{\link{assess_convergence}}
+#' @param C Number of cluster
+#' @param colnames A vector of item names. If NULL, we generate generic names for the items in the ranking.
+#' @param items Either a vector of item names, or a
+#'   vector of indices. If NULL, five items are selected randomly.
+#' @export
```

Comment:
Same comment as above, wouldn't it make sense to stop exporting these underlying functions now?

Reply:
I'm a bit divided myself. I'll mark this as resolved and take the conversation at https://github.com/ocbe-uio/BayesMallows/pull/263#discussion_r1032828173 as applicable to this change as well.